storage/adapter.go

```diff
@@ -0,0 +1,407 @@
+	}
+
+	pci, found := state.PreCommittedSectors[fmt.Sprint(sectorID)]
+	if found {
+		// TODO: If not expired yet, we can just try reusing sealticket
+		return pci.Info.CommR, true, nil
+	}
+
+	return nil, false, nil
+}
+
+func (m *StorageMinerNodeAdapter) GetSealSeed(ctx context.Context, preCommitMsgCid cid.Cid, interval uint64) (<-chan s2.SealSeed, <-chan s2.SeedInvalidated, <-chan s2.FinalityReached, <-chan *s2.GetSealSeedError) {
+	ready := make(chan s2.SealSeed)
+	invalidated := make(chan s2.SeedInvalidated)
+	finalized := make(chan s2.FinalityReached)
+	errs := make(chan *s2.GetSealSeedError)
+
+	// would be ideal to just use the events.Called handler, but it wouldnt be able to handle individual message timeouts
+
+	go func() {
+		mw, err := m.api.StateWaitMsg(ctx, preCommitMsgCid)
+		if err != nil {
+			err = xerrors.Errorf("failed to wait for pre-commit message: %+v", err)
+			errs <- s2.NewGetSealSeedError(err, s2.GetSealSeedFailedError)
+			return
+		}
+
+		if mw.Receipt.ExitCode != 0 {
+			log.Error("sector precommit failed: ", mw.Receipt.ExitCode)
+			err := xerrors.Errorf("sector precommit failed: %d", mw.Receipt.ExitCode)
+			errs <- s2.NewGetSealSeedError(err, s2.GetSealSeedFailedError)
+			return
+		}
+
+		n := mw.TipSet.Height() + build.InteractivePoRepDelay
+
+		// waiting for a seed to become available, e.g. mined into a block at
+		// the provided confidence level-block height
+		err = m.events.ChainAt(func(ectx context.Context, ts *types.TipSet, curH uint64) error {
+			randHeight := n - 1 // -1 because of how the messages are applied
+			rand, err := m.api.ChainGetRandomness(ectx, ts.Key(), int64(randHeight))
+			if err != nil {
+				err = xerrors.Errorf("failed to get randomness for computing seal proof: %w", err)
+				errs <- s2.NewGetSealSeedError(err, s2.GetSealSeedFatalError)
+				return err
+			}
+
+			ready <- s2.SealSeed{
+				BlockHeight: randHeight,
+				TicketBytes: rand,
+			}
+
+			return nil
+		}, func(ctx context.Context, ts *types.TipSet) error {
+			log.Warn("revert in interactive commit sector step")
+			// TODO: need to cancel running process and restart...
+			invalidated <- s2.SeedInvalidated{}
+
+			return nil
+		}, build.InteractivePoRepConfidence, n)
+		if err != nil {
+			log.Warn("waitForPreCommitMessage ChainAt errored: ", err)
+		}
+
+		// waiting for the seed to become final, i.e. mined into a block at
+		// finality
+		err = m.events.ChainAt(func(ectx context.Context, ts *types.TipSet, curH uint64) error {
+			finalized <- s2.FinalityReached{}
+			return nil
+		}, func(ctx context.Context, ts *types.TipSet) error {
+			// noop
+			return nil
+		}, build.Finality, n)
+
+		if err != nil {
+			log.Warn("waitForPreCommitMessage ChainAt errored: ", err)
+		}
+	}()
+
+	return ready, invalidated, finalized, errs
+}
+
+func (m *StorageMinerNodeAdapter) CheckPieces(ctx context.Context, sectorID uint64, pieces []s2.Piece) *s2.CheckPiecesError {
```

Comment:
I'd keep the check functions in the storage-miner module, and have the lower-level calls they make (such as `StateMarketStorageDeal`) on this interface. (optimize for least amount of logic here)